src/domain/service.ts

```diff
@@ -528,6 +534,101 @@ export const searchCollaboratorApplications = async (identity: Identity) => {
   );
 };
 
+export const searchPauseableApplications = async (currentDate: Date) => {
+  const {
+    durations: {
+      attestation: { count, unitOfTime },
+    },
+  } = await getAppConfig();
+  // find all apps that are APPROVED with an approval date matching the configured time period
+  // default is 1 year to match DACO but we will need this for testing
+  const approvalDate = moment(currentDate).subtract(
+    count,
+    unitOfTime as unitOfTime.DurationConstructor,
+  );
+
+  const approvalDayStart = moment(approvalDate).startOf('day').toDate();
+  const approvalDayEnd = moment(approvalDate).endOf('day').toDate();
+
+  // can still run this multiple times in the same 24hr period,
+  // as any apps that were paused previously will be ignored because we're looking for APPROVED state
+  const apps = await ApplicationModel.find({
+    state: 'APPROVED',
+    approvedAtUtc: {
+      $gte: approvalDayStart, // check for an approvedAtUtc within a 24hr period
+      $lte: approvalDayEnd,
+    },
+    attestedAtUtc: { $exists: false }, // check the applicant has not already attested
+  }).exec();
+
+  return apps;
+};
+
+const addToReport = (type: keyof Report, report: Report, appId: string) => {
+  report[type].count++;
+  report[type].ids.push(appId);
+  return report;
+};
+
+const pauseApplication = async (currentApp: Application, identity: Identity, reason?: string) => {
+  // set app in state
+  const appObj = new ApplicationStateManager(currentApp);
+  // update app state, including transition to paused + update event
+  const role = await getDacoRole(identity);
+  logger.info(`Role ${role} is trying to PAUSE appId ${currentApp.appId}`);
+  const result = appObj.updateApp({ state: 'PAUSED', pauseReason: reason }, false, {
+    id: identity.userId,
+    role,
+  });
+  // save new app state in db
+  await ApplicationModel.updateOne({ appId: result.appId }, result);
+  // retrieve updated app from db
+  const updatedApp = await ApplicationModel.findOne({
+    appId: result.appId,
+  }).exec();
+  return updatedApp?.toObject() as Application;
+};
+
+export const runPauseAppCheck = async (
+  emailClient: nodemail.Transporter<SMTPTransport.SentMessageInfo>,
+  report: Report,
+  user: Identity,
+  currentDate: Date,
+) => {
+  const pauseableApps = await searchPauseableApplications(currentDate);
+  const config = await getAppConfig();
+  if (pauseableApps.length === 0) {
+    logger.info('No applications need to be paused at this time.');
+    return report.pausedApps;
+  }
+
+  await Promise.all(
```

Comment:
if one promise fails the whole process fails check [Promise.allSettelled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)

Reply:
you could also use `updateMany` which can bulk update stuff